### ISSUE TITLE
Can't open Swift realm databases with Browser

### ISSUE THREAD

**Issue by mark-anders** [realm/realm-browser-osx#47]
I'm just getting started with Realm swift using the swift-2.0 branch. I've both successfully written a program that can write to and read from a database as well as run the TableView sample. In both cases, when I attempt to open the default.realm file, which is located in my simulator directory, I get an "Invalid database" error. 

Since I haven't been able to figure out how to upload the .realm files, here are links to them on Dropbox:

https://www.dropbox.com/s/u1dznkm77wtoh12/TableView.realm?dl=0

https://www.dropbox.com/s/6ps10qpbzt6uek3/MyApp.realm?dl=0

**Comment by TimOliver** [comment]
Hi Mark!

Thanks for moving the issue over here from the realm-cocoa repo!

I just had a look at those .realm files and they don't seem right. 4KB is too small, and when I opened them up in a hex editor, half of the bits in there were all zeros.

I'm getting the feeling that it's not specifically a problem with Realm Browser (Which is presently using the latest version of Realm Objective-C), but it looks like something might be happening in your app that's causing writes to the Realm files to become incomplete.

Are you able to post any of your app's Realm code here so I can see if there's anything that might be causing that?

Thanks!

**Comment by mark-anders** [comment]
Hi Tim!  Here are 2 links.  The first is to a test project that illustrates the problem.  Unzip it, run "pod install" and open the workspace.  It's pretty simple. All the code is in the App Delegate.

I've also included a link to database that it created, which like the others is invalid.

https://dl.dropboxusercontent.com/u/48424008/Realmbug.zip

https://dl.dropboxusercontent.com/u/48424008/Realmbug.realm

Hope that helps!

**Comment by thejojo87** [comment]
I have the same problem.swift2.0 realmswift .
i can read and query in the code ,but can not open the realm file.

**Comment by TimOliver** [comment]
Thanks a lot for that guys! I had a chat with the other engineers and then just confirmed the build for myself.

You might have uncovered a case we hadn't discovered yet where the Swift 2.0 branch of Realm is producing .realm files that are sufficiently different that the Objective-C version of Realm inside the Realm Browser is unable to read them.

Thanks for letting us know about this! We'll be following this up to ensure that the .realm files produced by the different bindings are kept consistent!

(On a sidenote, once you've updated to Xcode 7 beta 4, make sure you clear out and re-get the Swift 2.0 branch from CocoaPods. That sample app didn't initially work for me because of some new changes in Swift!)

**Comment by Fab1n** [comment]
Any progress on that? We have the same issue there!!!

**Comment by jpsim** [comment]
tl;dr; we're working to cut an official release of core as soon as possible with the new file format disabled

I posted an update on this yesterday in the realm-cocoa mailing list: https://groups.google.com/d/msg/realm-cocoa/nllwT4KYMzw/clEGScpW8McJ

> the `master` and `swift 2.0` branches of realm-cocoa are currently using a release of Realm's core which introduces a new file format (which supports null values) which can't be read by previous versions of Realm.

> Since the Realm Browser is using the latest released version of Realm, it can't read these newer files, detecting them as invalid databases instead. So your file is likely a valid database, but simply at a later file format that the browser can't read.

> We're working on cutting a new release of core disabling this new file format because we're not quite ready to release it officially. It was a mistake on our part to merge this version of core into the binding's master branch, which we're working to address as soon as possible.

> In the mean time, we don't recommend using Realm other than through official releases in production scenarios. This includes the Swift 2.0 branch.

**Comment by schickling** [comment]
+1

**Comment by jpsim** [comment]
As of https://github.com/realm/realm-cocoa/pull/2309 (~4 days ago), including the official [0.94 release](https://realm.io/news/realm-objc-swift-0.94/), Realm Swift and Realm Objective-C have reverted to the old file format, compatible with the Realm Browser. So this should no longer be a problem.

**Comment by ZAPHIT** [comment]
+1

**Comment by TimOliver** [comment]
@ZAPHIT

Are you experiencing this issue? What version of Realm are you running?

**Comment by ZAPHIT** [comment]
Yes.. i tried cocoa pods on realm.. i have updated xcode 7.1 .. cant open Realm file. says its has wrong format or encrypted, i have removed the encryption but still it cannot open.

**Comment by ZAPHIT** [comment]
the latest..

**Comment by TimOliver** [comment]
Oh okay! That would be Realm 0.96.1 then.

The current Realm Browser on the App Store can only open files up to v0.95. If you want, there's a pre-release version of the Browser that can open Realm files of version 0.96 and above here:

https://github.com/realm/realm-browser-osx/releases/tag/0.96-prerelease

I'm working on preparing a new build of the Browser that supports 0.96 (and a bunch of bug fixes for El Capitan) and sending it to the App Store this weekend.

**Comment by ZAPHIT** [comment]
thanks!..

**Comment by akmalazlanjamal** [comment]
I'm having the same problem with Realm 0.98.5. I use realm with Java.

**Comment by TimOliver** [comment]
@akmalazlanjamal Which version of the Browser are you using? We just released a new version (0.101.0) that introduced a file format change.

**Comment by akmalazlanjamal** [comment]
@TimOliver Thanks for the update. Now I can read after update Browser to the latest version.
